homeassistant/components/media_player/yamaha.py

```diff
@@ -108,4 +148,10 @@ def turn_on(self):
 
     def select_source(self, source):
         """Select input source."""
-        self._receiver.input = source
+        if self._reverse_mapping is not None:
+            if source in list(self._reverse_mapping.keys()):
```

Comment:
Is `self._reverse_mapping` a dict? If so, dicts implement the `__contains__()` method, which makes the in operator work.

This is much better python for line 152:

    if source in self._reverse_mapping: